x-pack/plugins/canvas/public/components/workpad_templates/workpad_templates.js

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiBasicTable,
+  EuiButtonIcon,
+  EuiPagination,
+  EuiSpacer,
+  EuiToolTip,
+  EuiHealth,
+  EuiButtonEmpty,
+  EuiSearchBar,
+} from '@elastic/eui';
+import { get, sortByOrder } from 'lodash';
+import { palettes } from '@elastic/eui/lib/services';
+import { getId } from '../../lib/get_id';
+import { Paginate } from '../paginate';
+import { templates } from './templates';
+
+const uniqueTags = templates
+  .reduce((acc, { tags = [] }) => {
+    tags.forEach(tag => {
+      if (acc.indexOf(tag) === -1) acc.push(tag);
+    });
+    return acc;
+  }, [])
+  .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
+
+const { colors } = palettes.euiPaletteColorBlind;
+
+const tagColorMapping = uniqueTags.reduce((acc, tag, i) => {
+  acc[tag] = colors[i % Object.keys(colors).length];
```

Comment:
How about just having a tags registry with hard coded color mappings?

Reply:
Not like people can add tags right now anyway. With a registry we can make tags pluggable, or load them from Elasticsearch later.